### DISCUSSION TITLE
How clone, duplicate current selected object or group of objects.

### DISCUSSION THREAD

**Discussion by nikmauro** [fabricjs/fabric.js#10247] [General]
Hello can you provide a simple code how to duplicate current object selection?

here my code

` duplicateSelectedObject() {

        const activeObject = this.canvas.getActiveObject();

        if (activeObject) {
            activeObject.clone(function(clonedObj) {
                // Offset the position of the duplicated object
                clonedObj.set({
                    left: activeObject.left + 20,
                    top: activeObject.top + 20,
                    evented: true
                });

                // Add the cloned object to the canvas
                this.canvas.add(clonedObj);
                this.canvas.setActiveObject(clonedObj);
                this.canvas.requestRenderAll();
            });
        } else {
            console.log("No object is currently selected.");
        }
    }`
Here the errors

```
Text.ts:1752 Uncaught TypeError: t is not iterable
    at Fo.toObject (Text.ts:1752:49)
    at Fo.clone (Object.ts:1240:29)
    at canvasApp.duplicateSelectedObject (canvas.js:348:26)
    at HTMLAnchorElement.<anonymous> (canvas.js:196:50)
    at HTMLAnchorElement.dispatch (jquery.min.js:2:40035)
    at v.handle (jquery.min.js:2:38006)
```

**Comment by nikmauro** [comment]
also try and this

```
 duplicateSelectedObject() {
        const activeObject = this.canvas.getActiveObject();
        if (activeObject) {
            const activeObjectData = activeObject.toObject();
            fabric.util.enlivenObjects([activeObjectData], (objects) => {
                objects.forEach((o) => {
                    o.set({
                        id:100,
                        top: o.top + 15,
                        left: o.left + 15
                    });
                    this.canvas.add(o);
                });
                this.canvas.setActiveObject(objects[0]);
                this.updateControls(); // Update controls after duplication
                this.canvas.requestRenderAll();
            });
        } else {
            console.warn("No object selected for duplication.");
        }
    }
```

**Comment by mhieupham1** [comment]
i think the object needs to be loaded before set active object, trying to use settimeout

**Comment by mhieupham1** [comment]
@nikmauro

**Comment by zhe-he** [comment]
[clone demo](https://codepen.io/zhe-he-the-vuer/pen/ZEdoBxj)

**Comment by asturur** [comment]
Are you using fabric 6 with fabric 5 examples? that could be it.
in fabric6 close is a promise and does not take a callback
